fml/raster_thread_merger.cc

```diff
@@ -17,23 +17,37 @@ RasterThreadMerger::RasterThreadMerger(fml::TaskQueueId platform_queue_id,
       gpu_queue_id_(gpu_queue_id),
       task_queues_(fml::MessageLoopTaskQueues::GetInstance()),
       lease_term_(kLeaseNotSet) {
-  is_merged_ = task_queues_->Owns(platform_queue_id_, gpu_queue_id_);
+  bool is_merged = task_queues_->Owns(platform_queue_id_, gpu_queue_id_);
+  is_merged_.store(is_merged);
+  if (is_merged) {
+    lease_term_ = 1;
+  }
 }
 
 void RasterThreadMerger::MergeWithLease(size_t lease_term) {
   FML_DCHECK(lease_term > 0) << "lease_term should be positive.";
-  if (!is_merged_) {
-    is_merged_ = task_queues_->Merge(platform_queue_id_, gpu_queue_id_);
+  if (!is_merged_.load()) {
+    FML_DLOG(ERROR) << "--- tm_: merged";
+    is_merged_.store(task_queues_->Merge(platform_queue_id_, gpu_queue_id_));
     lease_term_ = lease_term;
   }
 }
 
+void RasterThreadMerger::UnMergeNow() {
+  FML_CHECK(IsOnRasterizingThread());
+  lease_term_ = 0;
+  bool success = task_queues_->Unmerge(platform_queue_id_);
+  FML_CHECK(success) << "Unable to un-merge the raster and platform threads.";
+  is_merged_.store(false);
+  FML_DLOG(ERROR) << "--- tm_: unmerged";
```

Comment:
is this an error?

Reply:
No, I'm just logging it for testing purpose. Will remove before landing.